tests/test_beddays.py

```diff
@@ -82,6 +82,28 @@ def test_beddays_in_isolation(tmpdir, seed):
     hs.bed_days.remove_beddays_footprint(2)
     assert ([cap_bedtype1] * days_sim == tracker.values).all()
 
+    # 4) check that combining footprints for a person returns the expected output
+    # Test uses example fail case given in https://github.com/UCL/TLOmodel/issues/1399
+    # Person p has: bedtyp1 for 2 days, bedtype2 for 0 days.
+    # Person p then assigned: bedtype1 for 1 days, bedtype2 for 6 days.
+    # EXPECT: p's footprints are combined into bedtype1 for 2 days, bedtype2 for 5 days.
+    existing_footprint = {"bedtype1": 2, "bedtype2": 0}
+    incoming_footprint = {"bedtype1": 1, "bedtype2": 6}
+    expected_resolution = {"bedtype1": 2, "bedtype2": 5}
```

Comment:
Are we suggesting that we will have a max of two bed type in a footprint? if not, can we also look at how this function will behave when a footprint containing three bed types is issued? Its a rare event of-course but I also understand we are allowed to request bed days from all bed types expect non-bed space.

Reply:
The `BedDays.combine_footprints_for_same_patient` function should be general enough to handle any number of `bed_types`; it [loops through the `self.bed_types`](https://github.com/UCL/TLOmodel/blob/dfbc24c5cfda04f3657fbab517492e908c8dfcca/src/tlo/methods/bed_days.py#L357-L363) attribute so it will consider every bed type defined in the simulation (and thus, in the footprints).

Though you're correct that we should consider a more general case than just 2 beds - I've added another test case that involves 3 bed types.